woocommerce/woocommerce-ios
# [REST API] Merge `WordPressOrgRequest` into `RESTRequest`

Closes: #8512 


## Description
`WordPressOrgRequest` and `RESTRequest` have similar properties. For clarity we decided to merge them both and use 
 `RESTRequest`. 


## Testing instructions

- We need to test the areas in which `WordPressOrgRequest` (deleted in this PR) and `RESTRequest` are being used.
- Create a test store using JN. (without Jetpack)

*Jetpack installation*
- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- You should see a Jetpack error screen from where you can Install Jetpack or Connect Jetpack natively.
- Validate that the Jetpack installation/setup works as before.

*Application password*
1. Turn on the `applicationPasswordAuthenticationForSiteCredentialLogin` feature flag.
1. Login into the app using .org site credentials.
1. Navigate to the coupons tab.
1. Try creating, editing and deleting coupons.
1. All features in the coupons screen should work as before.

## Screenshots
NA

---
